packages/sdk/src/plt/v1/Token.ts

```diff
@@ -0,0 +1,254 @@
+/**
+ * Retrieves the balance of a token for a given account.
+ *
+ * @param {Token} token - The token to check the balance of.
+ * @param {AccountInfo | AccountAddress.Type} account - The account to check the balance for.
+ *
+ * @returns {Promise<BalanceOfResponse> | BalanceOfResponse} The balance of the token for the account.
+ */
+export function balanceOf(
+    token: Token,
+    account: AccountInfo | AccountAddress.Type
+): Promise<BalanceOfResponse> | BalanceOfResponse {
+    if (!AccountAddress.instanceOf(account)) {
+        return account.accountTokens.find((t) => t.id.symbol === token.info.id.symbol)?.state.balance;
+    }
+
+    return token.grpc.getAccountInfo(account).then((accInfo) => {
+        return accInfo.accountTokens.find((t) => t.id.symbol === token.info.id.symbol)?.state.balance;
+    });
+}
+
+/**
+ * Validates a token transfer.
+ *
+ * @param {Token} token - The token to transfer.
+ * @param {AccountAddress.Type} sender - The account address of the sender.
+ * @param {TokenTransfer | [TokenTransfer]} payload - The transfer payload.
+ *
+ * @returns {Promise<true>} A promise that resolves to true if the transfer is valid.
+ * @throws {InvalidTokenAmountError} If any token amount is not compatible with the token.
+ * @throws {InsufficientFundsError} If the sender does not have enough tokens.
+ * @throws {NotAllowedError} If a receiver is not allowed to receive tokens.
+ */
+export async function validateTransfer(
+    token: Token,
+    sender: AccountAddress.Type,
+    payload: TokenTransfer | [TokenTransfer]
+): Promise<true> {
+    const payloads = [payload].flat();
+
+    // Validate all amounts
+    payloads.forEach((p) => validateAmount(token, p.amount));
+
+    // Check the sender balance.
+    const senderBalance = (await balanceOf(token, sender)) ?? TokenAmount.zero(); // We fall back to zero, as the `token` has already been validated at this point.
+    const payloadTotal = payloads.reduce(
+        (acc, { amount }) => acc.add(TokenAmount.toDecimal(amount)),
+        TokenAmount.toDecimal(TokenAmount.zero())
+    );
+    if (TokenAmount.toDecimal(senderBalance).lt(payloadTotal)) {
+        throw new InsufficientFundsError(sender, TokenAmount.fromDecimal(payloadTotal));
+    }
+
+    // Check that all receivers are NOT on the deny list (if present), or that they are included in the allow list (if present).
+    const receiverPromises = payloads.map((p) => token.grpc.getAccountInfo(p.recipient));
+    const receivers = await Promise.all(receiverPromises);
+    receivers.forEach((r) => {
+        const accToken =
+            r.accountTokens.find((t) => t.id.symbol === token.info.id.symbol) ??
+            bail(new NotAllowedError(r.accountAddress));
+        if (accToken.state.memberDenyList || accToken.state.memberAllowList === false) {
+            throw new NotAllowedError(r.accountAddress);
+        }
```

Comment:
From my last information, the `denyList`/`allowList` will be checked on the `sender` address as well as the `receiver` address.

Reply:
Huh okay.. seems redundant in my head but will add it! Thanks